File: coinhive/src/main/java/ro/mdc_software/coinhive/CoinHiveFragment.java
package ro.mdc_software.coinhive;

import android.support.v4.app.Fragment;

/**
 * @author Cosmin Mihu
 */
public abstract class CoinHiveFragment extends Fragment implements CoinHive.Callback {

    private CoinHive.Miner coinHiveMiner;

    @Override
    public void onStart() {
        super.onStart();

        coinHiveMiner = new CoinHive.Miner(getActivity(), this);
    }

    public void stopMining() {
        coinHiveMiner.stopMining();
    }

    public void startMining() {
        coinHiveMiner.startMining();
    }

    @Override
    public void onDestroy() {
        if (CoinHive.getInstance().isLoggingEnabled()) {
            System.out.println("Mining stopped");
        }
        stopMining();
        super.onDestroy();
    }

    @Override
    public boolean isShowMining() {
        return false;
    }

    @Override
    public void onMiningStarted() {
    }

    @Override
    public void onMiningStopped() {
    }

    @Override
    public void onRunning(double hashesPerSecond, long totalHashes, long acceptedHashes) {
    }
}

File: coinhive/src/main/java/ro/mdc_software/coinhive/CoinHiveFragment.kt
package ro.mdc_software.coinhive

import android.support.v4.app.Fragment

/**
 * @author Cosmin Mihu
 */
abstract class CoinHiveFragment : Fragment(), CoinHive.Callback {

    private lateinit var coinHiveMiner: Miner

    override fun onStart() {
        super.onStart()

        coinHiveMiner = Miner(activity, this)
    }

    fun stopMining() {
        coinHiveMiner.stopMining()
    }

    fun startMining() {
        coinHiveMiner.startMining()
    }

    override fun onDestroy() {
        if (CoinHive.loggingEnabled) {
            println("Mining stopped")
        }
        stopMining()
        super.onDestroy()
    }

    override fun onMiningStarted() {}

    override fun onMiningStopped() {}

    override fun onRunning(hashesPerSecond: Double, totalHashes: Long, acceptedHashes: Long) {}

    override fun isShowMining() = false
}
